packages/socket/src/reconnectingsocket.spec.ts

```diff
@@ -1,34 +1,22 @@
-import assert from "assert";
-
 import { ReconnectingSocket } from "./reconnectingsocket";
 
 /** @see https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback */
-type Exec = (command: string, callback: (error: null | (Error & { readonly code: number })) => void) => void;
-
-let exec: Exec | undefined;
-let childProcessAvailable: boolean;
-
-try {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  exec = require("child_process").exec;
-  assert.strict(typeof exec === "function");
-  childProcessAvailable = true;
-} catch {
-  childProcessAvailable = false;
-}
+type Exec = (command: string, callback: (error: null | (Error & { readonly code?: number })) => void) => void;
+
+const getExec = async (): Promise<Exec> => {
+  const exec = (await import("child_process")).exec;
```

Comment:
We could catch the exception here and then just return `Exec | undefined`. This would probably lead to cleaner caller code